src/foundry/foundry.js/collections/documentCollections/compendiumCollection.d.ts

```diff
@@ -0,0 +1,206 @@
+    getIndex(): Promise<DocumentInstanceForMetadata<T>['data']['_source']>;
+
+    /**
+     * Get a single Document from this Compendium by ID.
+     * The document may already be locally cached, otherwise it is retrieved from the server.
+     * @param id -  The requested Document id
+     * @returns The retrieved Document instance
+     */
+    getDocument(id: string): Promise<DocumentInstanceForMetadata<T> | undefined>;
+
+    /**
+     * Load multiple documents from the Compendium pack using a provided query object.
+     * @param query - A database query used to retrieve documents from the underlying database
+     * @returns The retrieved Document instances
+     */
+    getDocuments(query?: Query): Promise<DocumentInstanceForMetadata<T>[]>;
+
+    /**
+     * Import a Document into this Compendium Collection.
+     * @param document - The existing Document you wish to import
+     * @returns The imported Document instance
+     */
+    importDocument(document: DocumentInstanceForMetadata<T>): Promise<DocumentInstanceForMetadata<T>>;
+
+    /**
+     * Fully import the contents of a Compendium pack into a World folder.
+     * @returns The imported Documents, now existing within the World
+     */
+    importAll({ folderId, folderName, options }: Partial<ImportAllOptions>): Promise<DocumentInstanceForMetadata<T>[]>;
+
+    /**
+     * Add a Document to the index, capturing it's relevant index attributes
+     * @param document -The document to index
+     */
+    indexDocument(document: DocumentInstanceForMetadata<T>): DocumentInstanceForMetadata<T>;
+
+    /**
+     * Create a new Compendium Collection using provided metadata.
+     * @param metadata - The compendium metadata used to create the new pack
+     * @param options - Additional options which modify the Compendium creation request
+     */
+    static createCompendium<T extends Metadata<any>>(
+      metadata: T,
+      options: Partial<DocumentModificationOptions>
+    ): Promise<CompendiumCollection<T>>;
+
+    /**
+     * Assign configuration metadata settings to the compendium pack
+     * @param settings - The object of compendium settings to define
+     * @returns A Promise which resolves once the setting is updated
+     */
+    configure(settings: Partial<CompendiumConfiguration>): Promise<void>;
+
+    /**
+     * Delete an existing world-level Compendium Collection.
+     * This action may only be performed for world-level packs by a Gamemaster User.
+     */
+    deleteCompendium(): Promise<this>;
+
+    /**
+     * Duplicate a compendium pack to the current World.
+     * @param label - A new Compendium label
+     */
+    duplicateCompendium({ label }: { label?: string }): Promise<this>;
+
+    /**
+     * Validate that the current user is able to modify content of this Compendium pack
+     * @remarks `requireUnlocked` has a default value of `true`
+     */
+    protected _assertUserCanModify({ requireUnlocked }: { requireUnlocked?: boolean }): boolean;
```

Comment:
Parameter is optional.
I guess the return type can be `true` instead of `boolean`.

Reply:
Agreed on the parameter, but I don't quite see why the return type should be true instead of a boolean

Reply:
I might have overseen something, but I do not see a possible `false` return here. With a `boolean` return, users might be misled to do things like `if(this._assertUserCanModify())` which is not how this method should be used.